libctru/source/services/dsp.c

```diff
@@ -0,0 +1,254 @@
+#include <3ds/types.h>
+#include <3ds/svc.h>
+#include <3ds/srv.h>
+#include <3ds/ipc.h>
+#include <3ds/services/dsp.h>
+
+Handle dspHandle = 0;
+
+
+Result dspInit(void)
+{
+	Result ret = 0;
+
+	if (dspHandle == 0)
+	{
+		ret = srvGetServiceHandle(&dspHandle, "dsp::DSP");
+		if (ret < 0) return ret;
+	}
+	if (ret < 0) return ret;
+	DSP_UnloadComponent();
+	return 0;
+}
+
+Result dspExit(void)
+{
+	Result ret = 0;
+//No need to call unload, it will be done automatically by closing the handle
+	if (dspHandle != 0)
+	{
+		ret = svcCloseHandle(dspHandle);
+		if (ret < 0) return ret;
+		dspHandle = 0;
+	}
+
+	return 0;
+}
+
+
+Result DSP_GetHeadphoneStatus(bool* is_inserted)
+{
+	Result ret = 0;
+	u32* cmdbuf = getThreadCommandBuffer();
+	cmdbuf[0] = IPC_MakeHeader(0x001F,0,0);
+	if ((ret = svcSendSyncRequest(dspHandle)) != 0) return ret;
+	*is_inserted = cmdbuf[2] & 0xFF;
```

Comment:
bool is a define for _Bool, and the standard says "When any scalar value is converted to _Bool, the result is 0 if the value compares equal to 0; otherwise, the result is 1." so I assume that it is not needed ?